### PR TITLE
[go] [cpd] Add CPD support for Antlr based grammar on Golang

### PR DESCRIPTION
**Pull request by matifraga**
Before submitting a PR, please check that:
 - [X] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [x] `./mvnw clean verify` passes. This will [build](https://github.com/pmd/pmd/blob/master/BUILDING.md) and test PMD, execute PMD and checkstyle rules. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**

Hi, here is team raptor again, we are adding CPD support for Golang based on an Antlr grammar. This is the first step in order to make Golang our first Antlr-based supported language for PMD rules. Our next steps towards that goal will include some changes over PMD to allow Antlr-based languages to be supported, and finally adding Golang as an example.

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-go/src/main/java/net/sourceforge/pmd/lang/go/GoLanguageModule.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+public class GoLanguageModule extends BaseLanguageModule {
+
+    /** The name. */
+    public static final String NAME = "Golang";
+    /** The terse name. */
+    public static final String TERSE_NAME = "go";
+
+    /**
+     * Create a new instance of Swift Language Module.
```
Comment (jsotuyod) [inline pmd-go/src/main/java/net/sourceforge/pmd/lang/go/GoLanguageModule.java]: copy paste from swift?

File: pmd-go/src/main/java/net/sourceforge/pmd/lang/go/GoLanguageModule.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+     */
+    public GoLanguageModule() {
+        super(NAME, null, TERSE_NAME, null, "go");
+        addVersion("", null, true);
```
Comment (jsotuyod) [inline pmd-go/src/main/java/net/sourceforge/pmd/lang/go/GoLanguageModule.java]: the current version of the go Language is `1` according to their docs: https://golang.org/doc/go1compat

**Comment by jsotuyod** [comment]
I'll fix as I merge

File: pmd-go/src/main/java/net/sourceforge/pmd/lang/go/GoLanguageModule.java (hunk 1 above)
Comment (matifraga) [reply pmd-go/src/main/java/net/sourceforge/pmd/lang/go/GoLanguageModule.java]: 🙈

**Pull request merged**